managed-token/program/src/lib.rs

```diff
@@ -220,3 +237,59 @@ pub fn process_close(accounts: &[AccountInfo]) -> ProgramResult {
     thaw(freeze_authority, mint, token_account, token_program, &seeds)?;
     close(token_account, dst_account, owner, token_program)
 }
+
+pub fn process_approve(accounts: &[AccountInfo], amount: u64) -> ProgramResult {
+    let Approve {
+        mint,
+        token_account,
+        owner,
+        upstream_authority,
+        delegate,
+        freeze_authority,
+        token_program,
+    } = Approve::load(accounts)?;
+    let seeds = get_authority_seeds_checked(upstream_authority.key, freeze_authority.key)?;
+    thaw(freeze_authority, mint, token_account, token_program, &seeds)?;
+    approve(token_account, owner, delegate, token_program, amount)?;
+    freeze(freeze_authority, mint, token_account, token_program, &seeds)
+}
+
+pub fn process_revoke(accounts: &[AccountInfo]) -> ProgramResult {
+    let Revoke {
+        mint,
+        token_account,
+        owner,
+        upstream_authority,
+        freeze_authority,
+        token_program,
+    } = Revoke::load(accounts)?;
+    let seeds = get_authority_seeds_checked(upstream_authority.key, freeze_authority.key)?;
+    thaw(freeze_authority, mint, token_account, token_program, &seeds)?;
+    revoke(token_account, owner, token_program)?;
+    freeze(freeze_authority, mint, token_account, token_program, &seeds)
+}
+
+pub fn process_wrap(accounts: &[AccountInfo]) -> ProgramResult {
+    let Wrap {
+        mint,
+        upstream_authority,
+        freeze_authority,
+        token_program,
+        mint_authority,
+    } = Wrap::load(accounts)?;
+    let (authority, _) = get_authority(upstream_authority.key);
+    set_authority(
+        mint,
+        &authority,
+        AuthorityType::MintTokens,
+        mint_authority,
+        token_program,
+    )?;
+    set_authority(
+        mint,
+        &authority,
+        AuthorityType::FreezeAccount,
+        freeze_authority,
+        token_program,
+    )
+}
```

Comment:
In general, I'm bearish about instructions that can be done outside of a program.  For example, this can be easily accomplished with a client library or CLI.

On the flip side, this is meant to be a "I don't want to deal with things, just do it for me" kind of program, so it seems like a better fit for this kind of instruction.

Reply:
agreed, can remove it for now to minimize the change.